plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -510,6 +515,49 @@ private ConnectorTableMetadata getTableMetadata(ConnectorSession session, Schema
             // This is just a catch-all solution so that we have any actionable information when eg. SELECT * FROM information_schema.columns fails.
             throw new RuntimeException("Failed to construct table metadata for table " + tableName, e);
         }
+        return Optional.ofNullable(getAvroSchemaUrl(tableMetadata.getProperties()))
+                .map(a -> rewriteMetadataFromAvroSchema(session, tableMetadata, a))
+                .orElse(tableMetadata);
+    }
+
+    private ConnectorTableMetadata rewriteMetadataFromAvroSchema(ConnectorSession session, ConnectorTableMetadata tableMetadata, String avroSchemaUrl)
+    {
+        Properties properties = new Properties();
+        properties.put(AVRO_SCHEMA_URL_KEY, avroSchemaUrl);
+        List<ColumnMetadata> avroColumnMetadata = new ArrayList<>();
+        try {
+            SchemaTableName schemaTableName = tableMetadata.getTable();
+            String schemaName = schemaTableName.getSchemaName();
+            String tableName = schemaTableName.getTableName();
+            TableLocation tableLocation = resolveTableLocation(session, tableMetadata, metastore.getTable(new HiveIdentity(session), schemaName, tableName));
+            Configuration conf = hdfsEnvironment.getConfiguration(new HdfsEnvironment.HdfsContext(session), tableLocation.getTargetPath());
+            AvroObjectInspectorGenerator aoig = new AvroObjectInspectorGenerator(
```

Comment:
Can we inject this code in `BridgingHiveMetastore#getTable` like we branch for avro table with schema url.